### PR TITLE
test: e2e test balancer (roundrobin upstream)

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

- Related issues
resolve https://github.com/apache/apisix-dashboard/issues/988

### CODE REVIEW

**Comment by idbeta** [comment]
@nic-chen when you are free, please help check the reason for this E2E failure.

**Comment by membphis** [comment]
no related issue?

**Comment by idbeta** [comment]
> no related issue?

fixed.

File: api/test/e2e/balancer_test.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+
+	//hit routes
```
Comment (membphis) [inline api/test/e2e/balancer_test.go:75]: code style

File: api/test/e2e/balancer_test.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		respBody, err = ioutil.ReadAll(resp.Body)
+		body := string(respBody)
+		if _, ok := res[body]; !ok {
+			res[body] = 1
+		} else {
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
+	assert.True(t, res["1982"] == res["1981"] && res["1981"] == res["1980"] && res["1980"] == 6)
```
Comment (membphis) [inline api/test/e2e/balancer_test.go:95]: code style

File: api/test/e2e/balancer_test.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+		respBody, err = ioutil.ReadAll(resp.Body)
+		body := string(respBody)
+		if _, ok := res[body]; !ok {
+			res[body] = 1
+		} else {
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
+	assert.True(t, res["1980"] == 3 && res["1981"] == 6 && res["1982"] == 9)
```
Comment (membphis) [inline api/test/e2e/balancer_test.go:144]: ditto

**Review by membphis — APPROVED** [review]
LGTM, except minor code style things

File: api/test/e2e/balancer_test.go (hunk 1 above)
Comment (idbeta) [reply api/test/e2e/balancer_test.go]: fixed.

File: api/test/e2e/balancer_test.go (hunk 2 above)
Comment (idbeta) [reply api/test/e2e/balancer_test.go]: fixed.

File: api/test/e2e/balancer_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/balancer_test.go]: fixed.

**Review by nic-chen — APPROVED** [review]
LGTM Except some code format

**Comment by juzhiyuan** [comment]
CI failed, @idbeta please take a look.

image

**Pull request merged**